File: mini-services/agent-factory/test-solver.ts
import { connectCdp, solveSliderCaptcha } from "./captcha-solver";

async function main() {
  const { browser, page } = await connectCdp();
  console.log("connected, page:", await page.url());
  await page.goto("https://chat.z.ai/", { waitUntil: "domcontentloaded" });
  await page.waitForTimeout(2500);

  // fill composer
  const ta = page.locator("textarea").first();
  await ta.fill("Reply with exactly: PING-OK");
  await page.waitForTimeout(600);

  // click send
  const sendBtn = page
    .locator('button[aria-label*="Send" i], button:has-text("Send")')
    .first();
  await sendBtn.click();
  await page.waitForTimeout(1600);

  const captchaVisible = await page.evaluate(
    () => !!document.querySelector(".window-show"),
  );
  console.log("captcha visible:", captchaVisible);

  if (captchaVisible) {
    const res = await solveSliderCaptcha(page);
    console.log("solver:", JSON.stringify(res));
  }

  await page.waitForTimeout(2500);
  const text = await page.evaluate(() => document.body.innerText.slice(0, 500));
  console.log("PAGE:", text.replace(/\n/g, " | "));
  const completions = [];
  browser.close();
  process.exit(0);
}

main().catch((e) => {
  console.error("FATAL", e);
  process.exit(1);
});

File: mini-services/agent-factory/test_solver.py
import json
import sys

sys.path.insert(0, "/home/z/my-project/mini-services/agent-factory")
from captcha_solver import connect, is_solved, solve_slider_captcha  # noqa: E402

pw, browser, page = connect("chat.z.ai")
print("connected, url:", page.url[:70])
page.goto("https://chat.z.ai/", wait_until="domcontentloaded")
page.wait_for_timeout(2600)

ta = page.locator("textarea").first
ta.fill("Reply with exactly: PING-OK")
page.wait_for_timeout(500)
page.locator('button[aria-label*="Send" i]').first.click()
page.wait_for_timeout(1700)

print("captcha visible:", not is_solved(page))
if not is_solved(page):
    res = solve_slider_captcha(page)
    print("solver:", json.dumps(res))

page.wait_for_timeout(3000)
print("PAGE:", page.evaluate("() => document.body.innerText.slice(0, 420)").replace("\n", " | "))
print("URL:", page.url)
pw.stop()
